diff --git a/cjs/extensions.js b/cjs/extensions.js
--- a/cjs/extensions.js
+++ b/cjs/extensions.js
@@ -26,24 +26,24 @@
         if (type) {
             applyExtensionObject(type, data.extensions);
             if (data.type === 'object' || data.type === 'interface') {
-                for (const [fieldName, fieldData] of Object.entries(data.fields)) {
+                for (const [fieldName, fieldData] of Object.entries(data.fields || {})) {
                     const field = type.getFields()[fieldName];
                     if (field) {
                         applyExtensionObject(field, fieldData.extensions);
-                        for (const [arg, argData] of Object.entries(fieldData.arguments)) {
+                        for (const [arg, argData] of Object.entries(fieldData.arguments || {})) {
                             applyExtensionObject(field.args.find(a => a.name === arg), argData);
                         }
                     }
                 }
             }
             else if (data.type === 'input') {
-                for (const [fieldName, fieldData] of Object.entries(data.fields)) {
+                for (const [fieldName, fieldData] of Object.entries(data.fields || {})) {
                     const field = type.getFields()[fieldName];
                     applyExtensionObject(field, fieldData.extensions);
                 }
             }
             else if (data.type === 'enum') {
-                for (const [valueName, valueData] of Object.entries(data.values)) {
+                for (const [valueName, valueData] of Object.entries(data.values || {})) {
                     const value = type.getValue(valueName);
                     applyExtensionObject(value, valueData);
                 }
